client/context/EditorContext.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { editObjectCatalogUrl, getObjectModelsDatastreamsUrl } from "../util/routes";
+import { useFetchContext } from "./FetchContext";
+
+/**
+ * Pass a shared entity to react components,
+ * specifically a way to make api requests.
+ */
+const editorContextParams = {
+    modelsCatalog: {},
+    currentPid: null,
+    currentModels: [],
+    currentDatastreams: [],
+    currentDatastream: null,
+    isDatastreamModalOpen: false,
+    datastreamModalState: null,
+};
+
+export const DatastreamModalStates = {
+    UPLOAD: "Upload",
+    VIEW: "View",
```

Comment:
Should you comment out all the states other than UPLOAD for now, and then uncomment them when they're implemented? It's a bit confusing that all the buttons appear but they all just upload for now. Might be worth hiding them until they're ready, if we want to do an incremental merge for just upload functionality.

Reply:
Made change.